### PR TITLE
Add readme for EF core Nuget Package 

### PR DESCRIPTION
**Pull request by lyndaidaii**
- [x] I've read the guidelines for [contributing](CONTRIBUTING.md) and seen the [walkthrough](https://youtu.be/9OMxy1wal1s?t=1869)
- [x] I've posted a comment on an issue with a detailed description of how I am planning to contribute and got approval from a member of the team
- [x] The code builds and tests pass locally (also verified by our automated build checks)
- [x] Commit messages follow this format:
```
        Summary of the changes
        - Add readme file for nuget package: https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/

        Fixes #bugnumber
```
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code follows the same patterns and style as existing code in this repo

Rendered README: https://github.com/dotnet/efcore/blob/32e73aebc1d64bbd519aaeb788547829ff173ecb/src/EFCore/README.md

Address: https://github.com/dotnet/efcore/issues/30467, https://github.com/dotnet/efcore/issues/26556

### CODE REVIEW

File: src/EFCore/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    // Add a new customer
+    context.Customers.Add(new Customer { Name = "John Doe" });
+    context.SaveChanges();
+
+    // Retrieve all customers
+    var customers = context.Customers.ToList();
+}
+```
+
+Microsoft.EntityFrameworkCore supports multiple database providers, including SQL Server, MySQL, PostgreSQL, SQLite, and others. You will need to install the provider package for your chosen database. For example, to use SQL Server, you would install the [Microsoft.EntityFrameworkCore.SqlServer package](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/7.0.3).
```
Comment (ajcvickers) [inline src/EFCore/README.md:44]: Add a link here to https://learn.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli

File: src/EFCore/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+## Additional documentation
+
+- [Getting Started with Entity Framework Core](https://learn.microsoft.com/en-us/ef/core/get-started/overview/first-app?tabs=netcore-cli).
+- Follow the [ASP.NET Core Tutorial](https://learn.microsoft.com/en-us/aspnet/core/data/ef-rp/intro?view=aspnetcore-7.0&tabs=visual-studio) to use EF Core in a web app.
+- [Releases and planning(roadmap)](https://learn.microsoft.com/en-us/ef/core/what-is-new/)
+- [How to write an EF Core provider](https://learn.microsoft.com/en-us/ef/core/providers/writing-a-provider)
+
+## Feedback
+
+If you have a specific question about using these projects, we encourage you to ask it on [Stack Overflow](https://stackoverflow.com/questions/tagged/entity-framework-core+or+entity-framework-core-2.1+or+entity-framework-core-3.1). If you encounter a bug or would like to request a feature, [submit an Github issue](https://github.com/dotnet/efcore/issues/new/choose). For more details, see [getting support](https://github.com/dotnet/efcore/blob/main/.github/SUPPORT.md).
```
Comment (ajcvickers) [inline src/EFCore/README.md:77]: We should update the StackOverflow tags here. @bricelam What tag(s) is/are best to use?

**Review by ajcvickers — APPROVED** [review]
Looks good to me. Just a couple of minor comments.

File: src/EFCore/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+You would then configure your database context to use the SQL Server provider:
+
+```c#
+using Microsoft.EntityFrameworkCore;
+
+public class MyDbContext : DbContext
+{
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+        optionsBuilder.UseSqlServer("Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=MyDatabase;");
```
Comment (bricelam) [inline src/EFCore/README.md:55]: Highly subjective, but I'd write it this way...
```suggestion
        optionsBuilder.UseSqlServer(@"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=MyDatabase");
```

File: src/EFCore/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
```
Comment (bricelam) [inline src/EFCore/README.md:56]: Highly subjective, but I write it this way...
```suggestion
    protected override void OnConfiguring(DbContextOptionsBuilder options)
        => options.UseSqlServer(@"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=MyDatabase");
```

**Comment by ajcvickers** [comment]
Superseded by #30934. Thanks @lyndaidaii for doing the work here,

**Pull request closed without merging**